src/main/java/ui/GUIController.java

```diff
@@ -0,0 +1,297 @@
+     * and their respective panels specify the display of metadata (through the UPDATED filter).
+     *
+     * Based on the hasMetadata state of the model, as specified by the event, the issues will then be displayed
+     * on their respective panels, or held back until their metadata requests have been fired and the downloaded
+     * metadata come back as a subsequent ModelUpdatedEvent (with e.hasMetadata being true).
+     *
+     * @param e The ModelUpdatedEvent triggered by the uiManager.
+     */
+    private void modelUpdated(ModelUpdatedEvent e) {
+        multiModel = e.model;
+
+        // Use updatedModel while handling a ModelUpdatedEvent to avoid race conditions.
+        IModel updatedModel = e.model;
+
+        // Set the new model to columnControl. This is in turn passed from ColumnControl to IssuePanel,
+        // down to each IssuePanelCard in order to display details about each issue such as labels and assignees.
+        columnControl.updateModel(updatedModel);
+
+        // Extracts all issues from the multimodel. This is then filtered through each of the panels' filters
+        // to produce the appropriate list of issues to be displayed.
+        ObservableList<TurboIssue> allModelIssues = FXCollections.observableArrayList(updatedModel.getIssues());
+
+        // Populated in processColumn calls.
+        HashMap<String, HashSet<Integer>> toUpdate = new HashMap<>();
+
+        columnControl.getChildren().forEach(child -> {
+            if (child instanceof IssueColumn) {
+                processColumn((IssueColumn) child, updatedModel, allModelIssues, toUpdate, e.hasMetadata);
+            }
+        });
+
+        // If toUpdate is empty, no metadata is requested.
+        dispatchMetadataRequests(toUpdate);
+    }
+
+    /**
+     * Handler method for an applyFilterExpression call from an IssueColumn, which is in turn triggered by
+     * the user pressing ENTER while the cursor is on the IssueColumn's filterTextField.
+     *
+     * The logic in this method is similar to that of modelUpdated, but only the IssueColumn whose
+     * filterTextField was changed will be processed.
+     *
+     * The multiModel to use here is stored from the last time a ModelUpdatedEvent was triggered.
+     *
+     * @param changedColumn The column whose filter expression had been changed by the user.
+     */
+    public void columnFilterExpressionChanged(IssueColumn changedColumn) {
+        ObservableList<TurboIssue> allModelIssues = FXCollections.observableArrayList(multiModel.getIssues());
+        HashMap<String, HashSet<Integer>> toUpdate = new HashMap<>();
+
+        // This is not triggered by a (metadata) update, so we pass false into the call.
+        processColumn(changedColumn, multiModel, allModelIssues, toUpdate, false);
+
+        dispatchMetadataRequests(toUpdate);
+    }
+
+    /**
+     * This method manages the flow of execution in filtering and updating a column.
```

Comment:
You can omit the `This method/class ...` part. It's ok to start `Manages the flow...`